File: elixir/acronym/acronym.exs
defmodule Acronym do
  @doc """
  Generate an acronym from a string.
  "This is a string" => "TIAS"
  """
  @spec abbreviate(String.t()) :: String.t()
  def abbreviate(string) do
    string
    |> String.split(~r/(?=\p{Lu})|[ -]/iu, trim: true)
    |> Enum.map(fn w -> w |> String.first() |> String.upcase() end)
    |> Enum.join()
  end
end

File: csharp/acronym/Acronym.cs
using System;
using System.Linq;

public static class Acronym
{
    public static string Abbreviate(string phrase) =>
        string.Concat(phrase.ToUpper().Split(new char[] { ' ', '-', '_' }, StringSplitOptions.RemoveEmptyEntries).Select(p => p.First()));
}